packages/menu-surface/index.tsx

```diff
@@ -0,0 +1,357 @@
+// The MIT License
+//
+// Copyright (c) 2018 Google, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+import * as React from 'react';
+import * as classnames from 'classnames';
+import {MDCMenuSurfaceFoundation, Corner} from '@material/menu-surface';
+
+export interface MenuSurfaceProps {
+  className: string,
+  anchorElement: HTMLElement,
+  anchorCorner: number,
+  anchorMargin: object,
+  styles: object,
+  coordinates: {
+    x?: number,
+    y?: number
+  },
+  onClose: () => void,
+  onOpen: () => void,
+  onKeyDown: (event: React.KeyboardEvent) => void,
+  quickOpen: boolean,
+  open: boolean,
+  fixed: boolean
+};
+
+type MenuSurfaceState = {
```

Comment:
I'd probably prefer using an `interface` over a `type` alias here, as they'll give you better TS compiler error messages (and allow for extension, but that's probably not particularly important here)

Reply:
1) You'd only need to export the state type if you expect people to extend it in some way, but generally I'd say it's an internal implementation detail.

2) I'd probably just export the base type (without the intersection). If you wanted to include the `div` props for anyone that was extending your component class, it might be better to use an `extends` rather than type intersection (e.g. `interface MenuSurfaceProps extends React.HTMLProps<HTMLDivElement> {}`)

Reply:
gotcha - I will only export the base props.